From 4cb65b354fca84b43b0b9fb803c904a729feea72 Mon Sep 17 00:00:00 2001
From: Kevin Peizner <kevin.peizner@racepointenergy.com>
Date: Fri, 4 Dec 2020 18:22:42 -0800
Subject: [PATCH] Fix refcount

---
 stdlib/public/SwiftShims/RefCount.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/public/SwiftShims/RefCount.h b/stdlib/public/SwiftShims/RefCount.h
index 095cd728..331ec5a8 100644
--- a/stdlib/public/SwiftShims/RefCount.h
+++ b/stdlib/public/SwiftShims/RefCount.h
@@ -1313,7 +1313,7 @@ class HeapObjectSideTableEntry {
 
   public:
   HeapObjectSideTableEntry(HeapObject *newObject)
-    : object(newObject), refCounts()
+    : object(newObject), refCounts(RefCounts<SideTableRefCountBits>::Initialized)
   { }
 
 #pragma clang diagnostic push
