From 4babf12695298d819e367479b9ce88c56fe434fd Mon Sep 17 00:00:00 2001
From: Kevin Peizner <kevin.peizner@racepointenergy.com>
Date: Thu, 8 Oct 2020 14:35:07 -0700
Subject: [PATCH] Ensure swift support is turned on

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75c419c..22cbd63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,9 @@ option(INSTALL_PRIVATE_HEADERS "installs private headers in the same location as
 option(ENABLE_SWIFT "enable libdispatch swift overlay" OFF)
 if(ENABLE_SWIFT)
   enable_language(Swift)
+  message("swift enabled" )
+else()
+  message( FATAL_ERROR "swift NOT enabled" )
 endif()
 
 option(ENABLE_THREAD_LOCAL_STORAGE "enable usage of thread local storage via _Thread_local" ON)
