From 1621c2cc4d75e6df6f6919acc18867e8c5412190 Mon Sep 17 00:00:00 2001
From: Luke Howard <lukeh@padl.com>
Date: Tue, 16 Sep 2025 17:28:03 +1000
Subject: [PATCH] build as dynamic library

Upstream-Status: Inappropriate [oe specific]
---
 Package.swift | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/Package.swift b/Package.swift
index 4360aab..16238ff 100644
--- a/Package.swift
+++ b/Package.swift
@@ -79,6 +79,7 @@ let package = Package(
     result.append(
       .library(
         name: "Testing",
+        type: .dynamic,
         targets: ["Testing"]
       )
     )
@@ -116,7 +117,7 @@ let package = Package(
       ],
       exclude: ["CMakeLists.txt", "Testing.swiftcrossimport"],
       cxxSettings: .packageSettings,
-      swiftSettings: .packageSettings + .enableLibraryEvolution(),
+      swiftSettings: .packageSettings + .enableLibraryEvolution("Testing"),
       linkerSettings: [
         .linkedLibrary("execinfo", .when(platforms: [.custom("freebsd"), .openbsd]))
       ]
@@ -186,7 +187,7 @@ let package = Package(
       dependencies: ["_TestingInternals",],
       exclude: ["CMakeLists.txt"],
       cxxSettings: .packageSettings,
-      swiftSettings: .packageSettings + .enableLibraryEvolution()
+      swiftSettings: .packageSettings + .enableLibraryEvolution("_TestDiscovery")
     ),
 
     // Cross-import overlays (not supported by Swift Package Manager)
@@ -196,7 +197,7 @@ let package = Package(
         "Testing",
       ],
       path: "Sources/Overlays/_Testing_CoreGraphics",
-      swiftSettings: .packageSettings + .enableLibraryEvolution()
+      swiftSettings: .packageSettings + .enableLibraryEvolution("_Testing_CoreGraphics")
     ),
     .target(
       name: "_Testing_Foundation",
@@ -208,7 +209,7 @@ let package = Package(
       // The Foundation module only has Library Evolution enabled on Apple
       // platforms, and since this target's module publicly imports Foundation,
       // it can only enable Library Evolution itself on those platforms.
-      swiftSettings: .packageSettings + .enableLibraryEvolution(.whenApple())
+      swiftSettings: .packageSettings + .enableLibraryEvolution("_Testing_Foundation", .whenApple())
     ),
 
     // Utility targets: These are utilities intended for use when developing
@@ -376,13 +377,9 @@ extension Array where Element == PackageDescription.SwiftSetting {
   ///   - condition: A build setting condition to apply to this setting.
   ///
   /// - Returns: A Swift setting that enables Library Evolution.
-  static func enableLibraryEvolution(_ condition: BuildSettingCondition? = nil) -> Self {
+  static func enableLibraryEvolution(_ target: String, _ condition: BuildSettingCondition? = nil) -> Self {
     var result = [PackageDescription.SwiftSetting]()
-
-    if buildingForDevelopment {
-      result.append(.unsafeFlags(["-enable-library-evolution"], condition))
-    }
-
+    result.append(.unsafeFlags(["-enable-library-evolution", "-enable-library-evolution", "-emit-module-interface", "-emit-module-interface-path", "\(target).swiftinterface"], condition))
     return result
   }
 
-- 
2.43.0

